socket_test.go

```diff
@@ -322,3 +322,68 @@ func TestConnMaxRetriesInfinite(t *testing.T) {
 		t.Fatalf("Dial called %d times, expected  at least %d", transport.dialCalledCount, atLeastExpectedRetries)
 	}
 }
+
+func TestSocketAutomaticReconnect(t *testing.T) {
+	listenEndpoint := "tcp://*:1234"
+	dialEndpoint := "tcp://localhost:1234"
+	message := "test"
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	wg := new(sync.WaitGroup)
```

Comment:
```suggestion
	var wg sync.WaitGroup
```

Reply:
I changed it, but there is one advantage of having a pointer to a WaitGroup instead of a WaitGroup: if it is not a pointer, someone could write code that copies the WaitGroup and get unexpected behavior. That is why I prefer `wg := new(sync.WaitGroup)` in general